qml/StyledToolBar.qml

```diff
@@ -53,6 +55,70 @@ ToolBar {
         return !!(navigator.currentItem && navigator.currentItem.objectName !== 'loadingView')
     }
 
+    function changeActiveKey(index, modelData) {
+        yubiKey.refreshDevicesDefault()
+        yubiKey.selectCurrentSerial(modelData.serial,
+                                    function (resp) {
+                                        if (resp.success) {
+                                            entries.clear()
+                                            yubiKey.currentDevice = modelData
+                                            yubiKey.calculateAll()
+                                        } else {
+                                            console.log("select device failed", resp.error_id)
+                                        }
+                                    })
+    }
+
+    function compareVersion(installed, current) {
+          var ver1 = installed.split('.').map( s => s.padStart(10) ).join('.');
+          var ver2 = current.split('.').map( s => s.padStart(10) ).join('.');
+          return ver1 < ver2;
+     }
+
+    XmlListModel {
+        id: xmlVersionModel
+        namespaceDeclarations: "declare default element namespace 'http://www.w3.org/2005/Atom';"
+        query: "/feed/entry"
+
+        XmlRole { name: "title"; query: "title/string()" }
+
+        onStatusChanged: {
+            if (status == XmlListModel.Ready && xmlVersionModel.count > 0) {
+                var title = xmlVersionModel.get(0).title;
+                var current = title.match(/\d+(\.\d+)+/g).join()
+                if (compareVersion(appVersion, current)) {
+                    navigator.confirm({
+                                "heading": qsTr("New version available!"),
+                                "description": qsTr("Yubico Authenticator %1 is now available, you have %2. Would you like to download it now?").arg(current).arg(appVersion),
```

Comment:
"current" suggests the currently installed version; let's call it `latest` instead?